protocols/sibyl_matrix.py

```diff
@@ -415,7 +436,15 @@ def new_user(self,user,typ=None,real=None):
   # @param pword (str) [None] the password for joining this Room
   # @return (Room) a new instance of this protocol's Room subclass
   def new_room(self,room_id_or_alias,nick=None,pword=None):
-    return MatrixRoom(self,room_id_or_alias,nick,pword)
+    if room_id_or_alias in self.room_aliases:
+      return self.room_aliases[room_id_or_alias]
+    else:
+      if room_id_or_alias.startswith("#"):
+        room = MatrixRoom(self,room_id_or_alias,nick,pword)
+        self.room_aliases[room_id_or_alias] = room
+        return room
+      else:
+        return MatrixRoom(self,room_id_or_alias,nick,pword)
```

Comment:
[OPTIONAL] This can be more compactly written:
```
def new_room(self,room_id_or_alias,nick=None,pword=None):
  if room_id_or_alias in self.room_aliases:
    return self.room_aliases[room_id_or_alias]

  room = MatrixRoom(self,room_id_or_alias,nick,pword)
  if room_id_or_alias.startswith("#"):
    self.room_aliases[room_id_or_alias] = room
  return room
```